### PR TITLE
(maint) Cleanup testrun script

### PR DESCRIPTION
**Pull request by Magisus**
This updates the various modes to take advantage of beaker subcommands.

### CODE REVIEW

File: acceptance/scripts/generic/testrun.sh [hunk 1]
```diff
@@ -14,62 +14,86 @@ export BEAKER_HELPER="${BEAKER_HELPER:-acceptance/lib/helper.rb}"
...
-  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
-  BEAKER="$BEAKER --config $BEAKER_CONFIG"
-  BEAKER="$BEAKER --preserve-hosts always"
-  $BEAKER
-  echo "Beaker exited with $?"
-  cp log/latest/hosts_preserved.yml .
+  BEAKER_INIT="$BEAKER_INIT --pre-suite $BEAKER_PRESUITE"
+  BEAKER_INIT="$BEAKER_INIT --tests $BEAKER_TESTSUITE"
+  BEAKER_INIT="$BEAKER_INIT --hosts $BEAKER_CONFIG"
+  BEAKER_INIT="$BEAKER_INIT --preserve-hosts always"
```
Comment (tvpartytonight) [inline acceptance/scripts/generic/testrun.sh]: Pretty sure you can get rid of this `--preserve-hosts` setting.

Comment (mwaggett) [reply acceptance/scripts/generic/testrun.sh]: Why?

Comment (Magisus) [reply acceptance/scripts/generic/testrun.sh]: Yes probably. None of the subcommnds except `destroy` clean up the machines, regardless of failure state.

Comment (mwaggett) [reply acceptance/scripts/generic/testrun.sh]: ah. so then could the other `--preserve-hosts` below be removed as well?

**Comment by Magisus** [comment]
Updated.

File: acceptance/scripts/generic/testrun.sh [hunk 2]
```diff
@@ -14,62 +14,85 @@ export BEAKER_HELPER="${BEAKER_HELPER:-acceptance/lib/helper.rb}"
...
-  BEAKER="$BEAKER --pre-suite $BEAKER_PRESUITE"
-  BEAKER="$BEAKER --hosts $BEAKER_CONFIG"
-  BEAKER="$BEAKER --pre-suite $BEAKER_PRESUITE"
-  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
-  BEAKER="$BEAKER --preserve-hosts onfail"
-  BEAKER="$BEAKER --debug --timeout 360"
-   $BEAKER
+  BEAKER_INIT="$BEAKER_INIT --pre-suite $BEAKER_PRESUITE"
+  BEAKER_INIT="$BEAKER_INIT --hosts $BEAKER_CONFIG"
+  BEAKER_INIT="$BEAKER_INIT --pre-suite $BEAKER_PRESUITE"
```
Comment (mwaggett) [inline acceptance/scripts/generic/testrun.sh]: This is duplicated.

File: acceptance/scripts/generic/testrun.sh [hunk 3]
```diff
@@ -14,62 +14,85 @@ export BEAKER_HELPER="${BEAKER_HELPER:-acceptance/lib/helper.rb}"
...
 
-  BEAKER="$BEAKER --config hosts_preserved.yml"
-  BEAKER="$BEAKER --preserve-hosts always"
-  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
-  $BEAKER
+  bundle exec beaker exec $BEAKER_TESTSUITE
   ;;
 
 
 * )
```
Comment (mwaggett) [inline acceptance/scripts/generic/testrun.sh]: I feel like there could be some DRY'ing up between this and the `-p` case? It took some scrolling up-and-down and comparing to figure out the difference between these cases. ¯\_(ツ)_/¯

**Comment by Magisus** [comment]
Updated, let me know what you think!

**Pull request merged**